ui/components/app/transaction-list-item-details/transaction-list-item-details.component.js

```diff
@@ -123,8 +135,26 @@ export default class TransactionListItemDetails extends PureComponent {
     });
   };
 
-  componentDidMount() {
-    const { recipientAddress, tryReverseResolveAddress } = this.props;
+  async componentDidMount() {
```

Comment:
Let's not make this function async. React will not "await" it. Instead, use a regular promise resolution pattern. In addition you should ensure that the component is still mounted prior to calling this.setState. There is no promise that the component will be mounted when your promise resolves, no matter how likely the scenario is, you might end up with situations where you are attempting to setState on an unmounted component. In a class component this is as easy as adding an instance variable this._mounted = true, in componentDidMount, and setting it to false in componentWillUnmount. When the promise resolves you check if its still true before call this.setState.

Reply:
Yap, well noticed, I'll address this change 👍🏼